scripts/utils/statisticsFunctions.ts

```diff
@@ -0,0 +1,127 @@
+import { Diff, DiffReport, Company, } from '../comparing-staging-production'
+
+export function reportStatistics(diffs: Diff[]) {
+//   const numbCorrectFieldsIncludeUndefined = diffs.reduce((acc: number, current: Diff) => {
+//     return current.productionValue === current.stagingValue ? acc+1 : acc; // this also captures if both are undefined
+//   }, 0);
+//   // How many of the fields are correct?
+//   const accuracy = {
+//     description: 'Out of all fields, how many are correct?',
+//     value: diffs.length > 0 ? numbCorrectFieldsIncludeUndefined / diffs.length : undefined,
+//     numbCorrectFieldsIncludeUndefined,
+//     numbFields: diffs.length
+//   }
+
+  // Out of all fields that are supposed to have a numerical value, How many are correct? (excludes all instances where prod has an undefined value)
+  const numbCorrectNumericalFields = diffs.reduce((acc: number, current: Diff) => { return !(current.productionValue === undefined && current.stagingValue === undefined) && (current.productionValue === current.stagingValue) ? acc + 1 : acc;}, 0)
+  const numbNumericalFields = diffs.reduce((acc: number, current: Diff) => { return !(current.productionValue === undefined && current.stagingValue === undefined) ? acc + 1 : acc;}, 0);
+  const accuracyNumericalFields = {
+    description: 'Out of all fields that are supposed to have a numerical value, how many are correct?',
+    value: numbCorrectNumericalFields/numbNumericalFields,
+    numbCorrectNumericalFields,
+    numbNumericalFields
+  }
+
+  const numbHasActualValueAndIsExtracted = diffs.reduce((acc: number, current: Diff) => { return (current.productionValue !== undefined && current.stagingValue !== undefined) ? acc + 1 : acc;}, 0);
+  const numbExtractedValues = diffs.reduce((acc: number, current: Diff) => { return current.stagingValue !== undefined ? acc + 1 : acc;}, 0);
+  const numbActualValues = diffs.reduce((acc: number, current: Diff) => { return current.productionValue !== undefined ? acc + 1 : acc;}, 0);
+
+  // Out of all the extracted values that are actually a value, how many of them are correct and how many are not?
+
+  // Out of all the extracted values by Garbo (staging), how many are actually a value and how many are made up by Garbo?
+  const precision = {
+    description: 'Out of all the extracted values by Garbo, how many are actually a value and how many are made up by Garbo?',
+    value: numbHasActualValueAndIsExtracted/numbExtractedValues,
+    numbHasActualValueAndIsExtracted,
+    numbExtractedValues
+  }
+
+  // Out of all the true values (prod), how many of them could Garbo extract a value for and how many could it not?
+  const recall = {
+    description: 'Out of all the true values, how many of them could Garbo extract a value for and how many could it not?',
```

Comment:
same with recall, here we are checking how often staging attempted to extract "something" where we actually expect "something". but we are not checking if it is correct which true recall would check..

recall:
`correct_values / correct_values + missed_values`